docs/src/operator/architecture.md

```diff
@@ -0,0 +1,65 @@
+# Node architecture
+
+The node itself consists of three distributed components: store, block-producer and RPC.
+
+The components can be run on separate instances when optimised for performance, but can also be run as a single process
+for convenience. At the moment both of Miden's public networks (testnet and devnet) are operating in single process
+mode.
+
+The inter-component communication is done using a gRPC API wnich is assumed trusted. In other words this _must not_ be
+public. External communication is handled by the RPC component with a separate external-only gRPC API.
+
+```mermaid
+---
+title: Infrastructure architecture example
+---
+graph LR;
+    
+  subgraph RPC
+  direction RL
+    rpc_a[RPC A]
+    rpc_b[RPC B]
+    rpc_i[......]
+    rpc_n[RPC N]
+  end
+
+  subgraph internal
+  direction TB
+    store
+    block-producer
+  end
+
+  load_balancer[Load Balancer]
+
+  load_balancer ---> RPC
+
+  RPC -- tx   ---> block-producer 
+  RPC -- query --> store
+
+  block-producer -- block --> store
+```
```

Comment:
I think a simple Excalidraw diagram (like we have in the readme) may be better here (we'd have more control of what to show and how). I've attached the source code for the diagrams I have.

[node design.excalidraw.zip](https://github.com/user-attachments/files/19153267/node.design.excalidraw.zip)

Reply:
I've used your image here instead.